packages/patternfly-4/react-core/src/components/Dropdown/InternalDropdownItem.js

```diff
@@ -0,0 +1,173 @@
+  /** Additional props are spread to the container component */
+  '': PropTypes.any // eslint-disable-line react/require-default-props
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  isHovered: false,
+  component: 'a',
+  role: null,
+  isDisabled: false,
+  href: '',
+  tooltip: null,
+  tooltipProps: {},
+  onClick: Function.prototype,
+  index: -1,
+  context: {
+    keyHandler: Function.prototype,
+    sendRef: Function.prototype
+  }
+};
+
+class DropdownItem extends React.Component {
+  ref = React.createRef();
+
+  componentDidMount() {
+    this.props.context.sendRef(this.props.index, this.ref.current, this.props.isDisabled);
+  }
+
+  onKeyDown = event => {
+    // Detected key press on this item, notify the menu parent so that the appropriate
+    // item can be focused
+    if (event.keyCode === KEY_CODES.TAB) return;
+    event.preventDefault();
+    if (event.keyCode === KEY_CODES.ARROW_UP) {
+      this.props.context.keyHandler(this.props.index, KEYHANDLER_DIRECTION.UP);
+    } else if (event.keyCode === KEY_CODES.ARROW_DOWN) {
+      this.props.context.keyHandler(this.props.index, KEYHANDLER_DIRECTION.DOWN);
+    } else if (event.keyCode === KEY_CODES.ENTER) {
+      if (!this.ref.current.getAttribute) {
+        ReactDOM.findDOMNode(this.ref.current).click(); // eslint-disable-line react/no-find-dom-node
+      } else {
+        this.ref.current.click && this.ref.current.click();
+      }
+    }
+  };
+
+  render() {
+    const {
+      className,
+      children,
+      isHovered,
+      context,
+      onClick,
+      component: Component,
+      isDisabled,
+      index,
+      href,
+      tooltip,
+      tooltipProps,
+      ...props
+    } = this.props;
+    const additionalProps = props;
+    let classes;
+    if (Component === 'a') {
+      additionalProps['aria-disabled'] = isDisabled;
+      additionalProps.tabIndex = isDisabled ? -1 : additionalProps.tabIndex;
+    } else if (Component === 'button') {
+      additionalProps.disabled = isDisabled;
+      additionalProps.type = additionalProps.type || 'button';
+    }
+
+    const renderWithTooltip = childNode => (
+      <Tooltip content={tooltip} {...tooltipProps}>
+        <span>{childNode}</span>
+      </Tooltip>
+    );
+
+    return (
+      <DropdownContext.Consumer>
+        {({ onSelect, itemClass, disabledClass, hoverClass }) => {
+          if (this.props.role === 'separator') {
+            classes = className;
+          } else {
+            classes = css(isDisabled && disabledClass, isHovered && hoverClass, className);
+          }
+          return (
+            <li role="none">
+              {React.isValidElement(children) ? (
+                React.Children.map(children, child => {
+                  const clonedElement = React.cloneElement(child, {
+                    className: css(
+                      isDisabled && disabledClass,
+                      isHovered && hoverClass,
+                      className,
```

Comment:
className should be on the `li` not on each of the children - `Classes applied to root element of dropdown item`

Reply:
I agree with you, but this is a carry over from the current Dropdown code
https://github.com/patternfly/patternfly-react/blob/master/packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js#L130
technically it could be considered a breaking change to move the className on the parent, although a pretty minor one. @dgutride should this wait for a breaking change release or ok to do here?

Reply:
@dgutride ^^

Reply:
if the api surface is removing anything on an existing component - we can't do it right now